control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -126,7 +126,7 @@ func (r *Reconciler) reconcileKind(ctx context.Context, channel *messagingv1beta
 	statusConditionManager.ConfigResolved()
 
 	// get the secret to access Kafka
-	secret, err := r.secret(ctx, channelConfigMap)
+	secret, err := security.Secret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: channelConfigMap}, r.SecretProviderFunc())
```

Comment:
I think for the channel we want a different `SecretLocator` since the `MTConfigMapSecretLocator` uses, as namespace, the namespace of the ConfigMap itself, while for the channel to be compatible with the existing one we need to read the namespace from the field of ConfigMap (I guess is called `authSecretNamespace`).

Reply:
KafkaChannel will also use the new configmap format.

Reply:
This means that the secret has to be in the `knative-eventing` namespace, is that ok (just wondering)?

Reply:
I guess for now it is good. We can improve this when we're working on namespace-scoped channel config (there are tickets already).